Components/Model/Repository/ProductStreams.php

```diff
@@ -0,0 +1,78 @@
+<?php
+/**
+ * Copyright (c) 2017, Nosto Solutions Ltd
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its contributors
+ * may be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * @author Nosto Solutions Ltd <shopware@nosto.com>
+ * @copyright Copyright (c) 2016 Nosto Solutions Ltd (http://www.nosto.com)
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD 3-Clause
+ */
+
+use Shopware\Models\Article\Article;
+
+/**
+ * Repository class for product streams.
+ *
+ * @package Shopware
+ * @subpackage Plugins_Frontend
+ */
+class Shopware_Plugins_Frontend_NostoTagging_Components_Model_Repository_ProductStreams
+{
+    /**
+     * @var \Doctrine\DBAL\Connection|mixed
+     */
+    private $conn;
+
+    /**
+     * Shopware_Plugins_Frontend_NostoTagging_Components_Model_Repository_ProductStreams constructor.
+     */
+    public function __construct()
+    {
+        $this->conn = Shopware()->Container()->get('dbal_connection');
+    }
+
+    /**
+     * Returns a list of Product Stream Selection Names
+     *
+     * @param Article $article
+     * @return array
+     */
+    public function getProductStreamsSelectionName(Article $article)
+    {
+        $builder = $this->conn->createQueryBuilder();
```

Comment:
Aren't there any methods or services to get this? I'm always a bit iof afraid of SQL and breaking schema changes.

Reply:
There is not. This is actually a 'join table' and Shopware CRUD operations are made with raw SQL queries in `Shopware/Controllers/Backend/ProductStream.php` as you can see in:
https://github.com/shopware/shopware/blob/5.4/engine/Shopware/Controllers/Backend/ProductStream.php#L190
And 
https://github.com/shopware/shopware/blob/5.4/engine/Shopware/Controllers/Backend/ProductStream.php#L203

Since we have no direct entity to the designated class, i had no other choice but to use raw SQL

Reply:
👍 Maybe a try catch?

Reply:
Try-catch block added